app/src/main/java/com/example/solidarity/EventDetailsActivity.java

```diff
@@ -84,5 +85,69 @@ protected void onCreate(Bundle savedInstanceState) {
         }
         tvGoing.setText(String.valueOf(currGoing) + " going");
 
+        btnLike.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (UserLikes == null) {
+                    UserLikes = new ArrayList<>();
+                    UserLikes.add(currentUser.getObjectId());
+                    event.setLikes(UserLikes);
+                    event.saveInBackground();
+                    currLikes += 1;
+                    tvLikes.setText(String.valueOf(currLikes) + " likes");
+                    btnLike.setBackgroundResource(R.drawable.like);
+                }
+
+                else if (!UserLikes.contains(currentUser.getObjectId())) {
+                    UserLikes.add(currentUser.getObjectId());
+                    event.setLikes(UserLikes);
+                    event.saveInBackground();
+                    currLikes += 1;
+                    tvLikes.setText(String.valueOf(currLikes) + " likes");
+                    btnLike.setBackgroundResource(R.drawable.like);
+                }
+                else {
+                    UserLikes.remove(currentUser.getObjectId());
+                    event.setLikes(UserLikes);
+                    event.saveInBackground();
+                    currLikes -= 1;
+                    tvLikes.setText(String.valueOf(currLikes) + " likes");
+                    btnLike.setBackgroundResource(R.drawable.like_empty);
+                }
+            }
+        });
+
+        btnGoing.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (goingList == null) {
+                    goingList = new ArrayList<>();
+                    goingList.add(currentUser.getObjectId());
+                    event.setGoingList(goingList);
+                    event.saveInBackground();
+                    currGoing += 1;
+                    tvLikes.setText(String.valueOf(currGoing) + " likes");
```

Comment:
Normally user facing texts should come from resource string for better internationalization. You don't have to do it now but you can put a TODO comment.